Makefile

```diff
@@ -117,6 +117,30 @@ define docker-up
 	fi
 endef
 
+# The dmake macro runs the current target in docker if and only if the
```

Comment:
@xiam @myleshorton This is the heart of the change.  This macro allows us to run any target either directly or in docker depending on the `USEDOCKER` environment variable.